### DISCUSSION TITLE
Proxmox VE 8.1

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#2095] [Announcements]
https://www.proxmox.com/en/about/press-releases/proxmox-virtual-environment-8-1

https://www.youtube.com/watch?v=0gjWxW2fMQs

If you've recently upgraded from Proxmox 8.0 to 8.1, there's a possibility of losing the No Nag function in case the `proxmox-widget-toolkit` is updated. To prevent this, it's recommended to re-run the Proxmox VE Post Install script. The script has been updated to accommodate the changes in the new versions of `proxmox-widget-toolkit`, ensuring the uninterrupted functionality of the No Nag feature.

**Comment by darthrater78** [comment]
Anything special to upgrade from 8.0?

**Reply by tteck** [reply]
Nope

**Comment by darthrater78** [comment]
https://pve.proxmox.com/pve-docs/chapter-pvesdn.html#pvesdn_installation

SDN will require some intervention after install for upgrades.

**Reply by tteck** [reply]
Are you using SDN?

**Reply by darthrater78** [reply]
I certainly may start now with these enhancements.

**Reply by tteck** [reply]
Keep us informed.

**Reply by tteck** [reply]
I just checked to see if `libpve-network-perl` was installed
```
root@server:~# dpkg -l | grep libpve-network-perl
ii  libpve-network-perl                  0.9.4                               all          Proxmox VE's SDN (Software Defined Network) stack
```
and it was. I've never used SDN 🤷🏻‍♂️

**Reply by darthrater78** [reply]
I merely parrot what the doc says. :) 

There was a user on the upgrade post on their forum who had an error and was pointed towards the article.

**Reply by tteck** [reply]
Yeah, I see now...
`dnsmasq` and `frr-pythontools` will need to be installed, along with `source /etc/network/interfaces.d/*`  added to `/etc/network/interfaces`. All in all, not bad.

```
apt-get install -y libpve-network-perl dnsmasq frr-pythontools
sed -i '$a\source /etc/network/interfaces.d/*' /etc/network/interfaces
```

**Comment by Skarnaut** [comment]
When installing pve 8.1 cleanly, the post-pve-install script does not disable subscription nag. Everything is fine in 8.0. Checked just now.

**Reply by tteck** [reply]
Did you delete your browser cache?

**Reply by Skarnaut** [reply]
Sure. I also logged in through another browser and in incognito modes

**Reply by tteck** [reply]
I'll take a look

**Reply by tteck** [reply]
Are you willing to test?

**Reply by Skarnaut** [reply]
yep

**Reply by tteck** [reply]
Great! Execute this in the shell to see if the nag is gone

```
sed -i 's/\.data\.status\.toLowerCase() !== '\''active'\''/\.data\.status\.toLowerCase() == '\''NoMoreNagging'\''/' /usr/share/javascript/proxmox-widget-toolkit/proxmoxlib.js
```

**Reply by Skarnaut** [reply]
It worked. The subscription notification is no longer displayed. Thank you.

**Reply by tteck** [reply]
not done yet...

**Reply by tteck** [reply]
To make it permanent, execute this in the shell

```
 echo "DPkg::Post-Invoke { \"dpkg -V proxmox-widget-toolkit | grep -q '/proxmoxlib\.js$'; if [ \$? -eq 1 ]; then { echo 'Removing subscription nag from UI...'; sed -i '/.*data\.status.*{/{s/\!//;s/active/NoMoreNagging/}' /usr/share/javascript/proxmox-widget-toolkit/proxmoxlib.js; }; fi\"; };" >/etc/apt/apt.conf.d/no-nag-script
```

**Reply by Skarnaut** [reply]
After entering the first command and clearing the cache, subscription nag disappeared. I checked several browsers and incognito mode. After that, I rebooted the pve node and subscription nag also did not appear. That is, the command works even after a reboot. Why do I need a second command in this case?

**Reply by tteck** [reply]
Because `/etc/apt/apt.conf.d/no-nag-script` has the script that doesn't work. When Proxmox updates `proxmoxlib.js` the Nag will reappear. You can test this now by reinstalling `proxmox-widget-toolkit`
```
apt --reinstall install proxmox-widget-toolkit
```

**Reply by Skarnaut** [reply]
When reinstalling proxmox-widget-toolkit in the console, I saw the following line:
```
root@pve:~# apt --reinstall install proxmox-widget-toolkit
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
0 upgraded, 0 newly installed, 1 reinstalled, 0 to remove and 0 not upgraded.
Need to get 210 kB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 http://download.proxmox.com/debian/pve bookworm/pve-no-subscription amd64 proxmox-widget-toolkit all 4.1.3 [210 kB]
Fetched 210 kB in 1s (234 kB/s)
(Reading database ... 73652 files and directories currently installed.)
Preparing to unpack .../proxmox-widget-toolkit_4.1.3_all.deb ...
Unpacking proxmox-widget-toolkit (4.1.3) over (4.1.3) ...
Setting up proxmox-widget-toolkit (4.1.3) ...
Removing subscription nag from UI...
```
Subscription nag is automatically deleted as seen in the last line. I assume this is due to the fact that I have already executed a post-pve-install script before performing the first fix

**Reply by tteck** [reply]
Yes, it thinks it's Removing subscription nag from UI, but you know that it isn't.

Proxmox VE Post Install script has been updated
